source/features/show-pr-file-state.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import {observe} from 'selector-observer';
+import elementReady from 'element-ready';
+import * as pageDetect from 'github-url-detection';
+
+import DiffAddedIcon from 'octicon/diff-added.svg';
+import DiffRemovedIcon from 'octicon/diff-removed.svg';
+import DiffModifiedIcon from 'octicon/diff-modified.svg';
+
+import features from '.';
+import fetchDom from '../helpers/fetch-dom';
+
+async function init(): Promise<false | void> {
+	const jumpList = await elementReady('details [src*="show_toc"]')!;
+	const src = jumpList!.getAttribute('src')!;
+	const fileList = await fetchDom(src, '[data-filterable-for]');
```

Comment:
I like that this is short, but if we preload that list in-place instead we don't _waste_ an HTTP request. I think `tags-dropdown` and `forked-to` do it via `mouseover`, but I don't remember. This `include-fragment` preloader could probably be made into a function.

Reply:
From what I see they are all using fetchdom